pandas/tests/extension/test_sparse.py

```diff
@@ -194,6 +194,19 @@ def test_reindex(self, data, na_value):
 # Skipping TestSetitem, since we don't implement it.
 
 
+class TestIndex(base.BaseIndexTests):
+    def test_index_from_array(self, data):
+        idx = pd.Index(data)
+        # TODO do we want to preserve the sparse dtype in the index
+        # now this is possible?
```

Comment:
So currently we still explicitly handle SparseArrays as input and ensure we use the subtype's Index class:

```
         elif isinstance(dtype, SparseDtype): 
             return cls._dtype_to_subclass(dtype.subtype) 
```

in

https://github.com/pandas-dev/pandas/blob/5ee8cf1daed481bdfd28399468a1233736332db3/pandas/core/indexes/base.py#L547-L571

I think that's the one case of an ExtensionArray which is not officially "experimental" that did not yet have its own Index subclass, and thus would give a change in behaviour. Not sure whether we want to preserve the current behaviour, or remove this special case and allow storing a SparseArray in the index.

Reply:
i think for 2.0 we'd want to remove the special-case.  if we try to get this in for 1.4 we'd need to deprecate for this particular case

Reply:
Yes, agreed, let's keep the SparseArray special case for now